src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/proxy/gui/ProxyControlGui.java

```diff
@@ -928,6 +988,46 @@ private JPanel createHTTPSamplerPanel() {
         prefixHTTPSampleName.addKeyListener(this);
         prefixHTTPSampleName.setName(PREFIX_HTTP_SAMPLER_NAME);
 
+        DefaultComboBoxModel<String> choiceNumbering = new DefaultComboBoxModel<>();
+        choiceNumbering.addElement(JMeterUtils.getResString("sample_numbering_mode_prefix")); // $NON-NLS-1$
+        choiceNumbering.addElement(JMeterUtils.getResString("sample_numbering_mode_suffix")); // $NON-NLS-1$
+        choiceNumbering.addElement(JMeterUtils.getResString("sample_numbering_mode_no_number")); // $NON-NLS-1$
+        String sNumberingMode = JMeterUtils.getPropDefault("proxy.number.mode", "prefix"); // $NON-NLS-1$
```

Comment:
This way of communicating through System property is not clean, it break object isolation and might make difficult to maintain.